### PR TITLE
Initial Commit for Text Area component

### PR DESCRIPTION
**Pull request by Parag27-cakewalk**
http://cakewalkuae/softstudio-projectmanagement#468

### CODE REVIEW

File: apps/www/registry/tui/ui/textarea.tsx [hunk 1]
```diff
@@ -1,21 +1,441 @@
...
+                              </>
+                            ) : (
+                              <>
+                                {icon ? <Icon name={icon} className={cn(`${iconStyle}`, className)} /> : null}
+                                <span className={cn("hidden truncate sm:ml-2 sm:block", className)}>{assignButtonName}</span>
+                              </>
+                            )}
+                          </button>
+                          {isListVisible && (
+                            <ul className={cn("absolute right-0 z-10 mt-1 max-h-56 w-52 overflow-auto rounded-lg py-3 text-base shadow sm:text-sm", className)}>
```
Comment (sohebcakewalk) [inline apps/www/registry/tui/ui/textarea.tsx:272]: @Parag27-cakewalk - Why fixed width and height?

File: apps/www/registry/tui/ui/textarea.tsx [hunk 2]
```diff
@@ -1,21 +1,441 @@
...
+                              </>
+                            ) : (
+                              <>
+                                {icon ? <Icon name={icon} className={cn(`${iconStyle}`,className)} /> : null}
+                                <span className={cn("hidden truncate sm:ml-2 sm:block", className)}>{labelButtonName}</span>
+                              </>
+                            )}
+                          </button>
+                          {isLabelListVisible && (
+                            <ul className={cn("absolute right-0 z-10 mt-1 max-h-56 w-52 overflow-auto rounded-lg py-3 text-base shadow ring-1 ring-opacity-5 focus:outline-none sm:text-sm", className)}>
```
Comment (sohebcakewalk) [inline apps/www/registry/tui/ui/textarea.tsx:304]: @Parag27-cakewalk  - Why fixed width and height?

File: apps/www/registry/tui/ui/textarea.tsx [hunk 3]
```diff
@@ -1,21 +1,441 @@
...
+                              </>
+                            ) : (
+                              <>
+                                {icon ? <Icon name={icon} className={cn(`${iconStyle}`, className)} /> : null}
+                                <span className={cn("hidden truncate sm:ml-2 sm:block", className)}>{dueDateButtonName}</span>
+                              </>
+                            )}
+                          </button>
+                          {isDueDateListVisible && (
+                            <ul className={cn("absolute right-0 z-10 mt-1 max-h-56 w-52 overflow-auto rounded-lg py-3 text-base shadow sm:text-sm", className)}>
```
Comment (sohebcakewalk) [inline apps/www/registry/tui/ui/textarea.tsx:339]: @Parag27-cakewalk - Why fixed width and height?

File: apps/www/registry/tui/ui/textarea.tsx [hunk 4]
```diff
@@ -1,21 +1,441 @@
...
+                          type="button" className={cn("inline-flex h-6 w-6 items-center justify-center rounded-full", className)}>
+                          {attachIcon ? <Icon name={attachIcon} className={cn(`${iconStyle}`, className)} /> : null}
+                        </button>
+                      </div>
+
+                      <div className={cn("flex root", className)}>
+                        {
+                          selectedIcon ? (
+                            <>
+                              <div className={cn(`bg-${selectedIcon.bg} flex h-8 w-8 items-center justify-center rounded-full`, className)}>
```
Comment (sohebcakewalk) [inline apps/www/registry/tui/ui/textarea.tsx:395]: @Parag27-cakewalk - Why fixed width and height?

File: apps/www/registry/tui/ui/textarea.tsx [hunk 5]
```diff
@@ -1,21 +1,441 @@
...
+                </div> : ""
+            }
+            {
+              showIconForTextArea ?
+                <>
+                  <div className={cn("flex justify-between pt-2", className)}>
+                    <div className={cn("flex items-center space-x-5", className)}>
+                      <div className={cn("flex root", className)}>
+                        <button
+                          type="button" className={cn("inline-flex h-6 w-6 items-center justify-center rounded-full", className)}>
```
Comment (sohebcakewalk) [inline apps/www/registry/tui/ui/textarea.tsx:386]: @Parag27-cakewalk - Why fixed width and height?

File: apps/www/registry/tui/ui/textarea.tsx [hunk 6]
```diff
@@ -1,21 +1,441 @@
...
+
+
+                      </div>
+                    </div>
+
+                    <div className={cn("flex-shrink-0", className)}>
+                      <button type="submit" className={cn("rounded-md px-3 py-1.5 font-medium text-base leading-6 ",buttonColor(color),className)}>{submitButtonName}</button>
+                    </div>
+                  </div>
+                  <ul className={cn("absolute z-10 -ml-6 w-60 rounded-lg text-base shadow focus:outline-none sm:ml-auto w-64 text-sm", className)}>
```
Comment (sohebcakewalk) [inline apps/www/registry/tui/ui/textarea.tsx:418]: @Parag27-cakewalk - Why fixed width and height?

**Review by sohebcakewalk — CHANGES_REQUESTED** [review]
Fix all the changes

File: apps/www/registry/tui/ui/textarea.tsx [hunk 7]
```diff
@@ -1,21 +1,441 @@
...
+    const [isDueDateListVisible, setIsDueDateListVisible] = useState(false);
+    const [selectedOption, setSelectedOption] = useState<AssignListItem | null>(null);
+    const [selectedLabelOption, setSelectedLabelOption] = useState<LabelListItem | null>(null);
+    const [selectedDueDateOption, setSelectedDueDateOption] = useState<DueDateListItem | null>(null);
+
+    const buttonColor = (color?: string) => {
+      return ` bg-${color}-600 text-white`
+    }
+
+    const handleCommentButtonClick = () => {
```
Comment (raghavcakewalk) [inline apps/www/registry/tui/ui/textarea.tsx:136]: wrap these in useCallbacks

Comment (raghavcakewalk) [reply apps/www/registry/tui/ui/textarea.tsx:183]: for example:  
   const handleWriteButtonClick = useCallback(() => {
      memoizedSetShowWrite(true);
      memoizedSetShowComment(false);
    }, []);
    
    It memoize the functions which returns the same result each time component renders.
    Make sure to momoize such functions or constants

**Review by sohebcakewalk — APPROVED** [review]

**Pull request merged**